NitroxClient/GameLogic/MobileVehicleBay.cs

```diff
@@ -44,6 +44,13 @@ public void BeginCrafting(GameObject constructor, TechType techType, float durat
                 packetSender.Send(beginCrafting);
 
                 vehicles.SpawnDefaultBatteries(constructedObject, childIdentifiers);
+
+                Optional<MonoBehaviour> monoBehaviour = Optional.OfNullable(constructor.GetComponent<MonoBehaviour>());
+                //We want to store the fallen position of the object to avoid flying object on reload 
+                if (monoBehaviour.HasValue)
```

Comment:
Simple implicit boolean check is fine for GetComponent.  Optional is mainly for return values.